### PR TITLE
Chore/ts entry

### PR DESCRIPTION
**Pull request by rishabh3112**
Working entry written in ts ⭐

### CODE REVIEW

**Comment by rishabh3112** [comment]
Lint errors else is fine @sendilkumarn

File: tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,62 @@
...
-      "./*",
-      "./utils/*"
-  ]
-}
+{
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
```
Comment (sendilkumarn) [inline tsconfig.json]: can we remove the commented lines?

**Comment by sendilkumarn** [comment]
@rishabh3112 can you fix it the lint errors?

File: tsconfig.json (hunk 1 above)
Comment (rishabh3112) [reply tsconfig.json]: Sure,
Will in next commit

**Comment by rishabh3112** [comment]
Check @sendilkumarn

**Comment by rishabh3112** [comment]
Should I remove `eslint` or let it be there @sendilkumarn?

File: generator.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  let outputDir;
+
+  this.devConfig.topScope = [
+  "const webpack = require('webpack')",
+  "const path = require('path')",
+  ];
+
+  welcomeMessage();
+  return this.prompt(questions.entryQuestion)
+  .then((entryAnswer) => {
```
Comment (AjayPoshak) [inline generator.ts]: I think that `entryAnswer` should also have a type.

File: generator.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    manifestDetails = {
+    description,
+    name,
+    shortName,
+    startURL,
+    themeColor,
+    };
+    return this.prompt([Confirm("favicon", "Do you have a existing Favicon to add ?")]);
+  })
+  .then((answer) => {
```
Comment (AjayPoshak) [inline generator.ts]: Same here, `answer` should have a type.

File: generator.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  * @returns {void}
+  */
+ public prompting(): Promise<void> {
+
+  const done = this.async();
+  let serviceWorker = false;
+  let manifestDetails = {};
+  let favPath;
+  let outputDir;
+
```
Comment (AjayPoshak) [inline generator.ts]: I am not sure, but I think that these variable declarations should also have types.

The rationale behind it is that suppose we expect `favPath` to be a string, but it gets reassigned as `boolean` mistakenly.  So this `TS` system should be able to flag that as an error.

File: generator.ts (hunk 2 above)
Comment (rishabh3112) [reply generator.ts]: type of `entryAnswer` is inherited from `yeoman-generator` i think 🤔

File: generator.ts (hunk 4 above)
Comment (rishabh3112) [reply generator.ts]: Here we should 😁
Thanks @ajayposhak 👍

**Comment by rishabh3112** [comment]
Have a look now @sendilkumarn

File: generator.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  "const webpack = require('webpack')",
+  "const path = require('path')",
+  ];
+
+  welcomeMessage();
+  return this.prompt(questions.entryQuestion)
+  .then((entryAnswer) => {
+    this.webpackOptions.entry = entryAnswer.entryFile;
+
+    return this.prompt([Confirm("serviceWorker", "Do you want to add Service Worker?")]);
```
[the remark this answers is not in the document]
Comment (bhavya9107) [reply generator.ts]: I think this was left in the PR for refactoring.
cc @AjayPoshak

I will have It done in seperate as it is out of scope for this PR.

**Comment by rishabh3112** [comment]
I think this was left in the PR for refactoring.
cc @AjayPoshak

I will have It done in seperate as it is out of scope for this PR.

**Comment by AjayPoshak** [comment]
@rishabh3112 My bad, I'll fix it.

File: generator.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
+import { Confirm, Input } from "@webpack-cli/webpack-scaffold";
+import Generator from "yeoman-generator";
+import createDevConfig from "./dev-config";
+import questions from "./questions";
+import getPackageManager from "./utils/package-manager";
+import welcomeMessage from "./utils/welcome";
+
+module.exports = class WebpackGenerator extends Generator {
+ [x: string]: any;
```
[the remark this answers is not in the document]
Comment (rishabh3112) [reply generator.ts]: index signature for public data members of classes

File: generator.ts [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+import questions from "./questions";
+import getPackageManager from "./utils/package-manager";
+import welcomeMessage from "./utils/welcome";
+
+module.exports = class WebpackGenerator extends Generator {
+
+ [x: string]: any;
+ public dependencies: string[];
+ public devConfig: any;
+ public webpackOptions: any;
```
Comment (sendilkumarn) [inline generator.ts]: There is no use in specifying `any` here. We can just create an interface that will help us a lot. A future enhancement for sure 👍

**Pull request merged**